classes/external/get_data_merge_requests.php

```diff
@@ -0,0 +1,151 @@
+                                    string $keepuservalue,
+                                    string $keepuserid,
+                                    int $id,
+                                    int $status) {
+        global $DB;
+        // Validate all of the parameters.
+        $params = array();
+        $params = self::validate_parameters(self::execute_parameters(),
+                                            ['removeuserfield' => $removeuserfield,
+                                            'removeuservalue' => $removeuservalue,
+                                            'removeuserid' => $removeuserid,
+                                            'keepuserfield' => $keepuserfield,
+                                            'keepuservalue' => $keepuservalue,
+                                            'keepuserid' => $keepuserid,
+                                            'id' => $id,
+                                            'status' => $status]);
+        $sql = "SELECT
+                    id, removeuserfield, removeuservalue, removeuserid,
+                    keepuserfield, keepuservalue, keepuserid,
+                    timeadded, timemodified, status, retries, log
+                FROM
+                   {" .merge_request::TABLE_MERGE_REQUEST. "}
+                ";
+        $paramsquery = array();
+        if (isset($removeuserfield) && !empty($removeuserfield)) {
+            $whereclauses[] = 'removeuserfield = ?';
+            array_push($paramsquery, $removeuserfield);
+        }
+        if (isset($removeuservalue) && !empty($removeuservalue)) {
+            $whereclauses[] = 'removeuservalue = ?';
+            array_push($paramsquery, $removeuservalue);
+        }
+        if (isset($removeuserid) && !empty($removeuserid)) {
+            $whereclauses[] = 'removeuserid = ?';
+            array_push($paramsquery, $removeuserid);
+        }
+        if (isset($keepuserfield) && !empty($keepuserfield)) {
+            $whereclauses[] = 'keepuserfield = ?';
+            array_push($paramsquery, $keepuserfield);
+        }
+        if (isset($keepuservalue) && !empty($keepuservalue)) {
+            $whereclauses[] = 'keepuservalue = ?';
+            array_push($paramsquery, $keepuservalue);
+        }
+        if (isset($keepuserid) && !empty($keepuserid)) {
+            $whereclauses[] = 'keepuserid = ?';
+            array_push($paramsquery, $keepuserid);
+        }
+        if (isset($id) && !empty($id)) {
+            $whereclauses[] = 'id = ?';
+            array_push($paramsquery, $id);
+        }
+        if (isset($status) && !empty($status)) {
+            $whereclauses[] = 'status = ?';
+            array_push($paramsquery, $status);
+        }
+        if (count($whereclauses) > 0) {
+            $sql .= 'WHERE ' . implode(' AND ', $whereclauses);
+        }
+        return $DB->get_records_sql($sql, $paramsquery);
+    }
+
+    public static function execute_returns() {
```

Comment:
I modified the code in a cleaner way.
Calling ws from postman now i receive this error:
{
    "exception": "invalid_response_exception",
    "errorcode": "invalidresponse",
    "debuginfo": "Error in response - Missing following required key in a single structure: id"
}
$sql and $paramquery are well defined

Reply:
Yeah, it's natural. You say in `execute_returns()` that you return a simple object. However, in `execute()` you return an array of objects. 

The real problem here is that we are informing in a wrong way the result type. We should return an array of elements, of tye type we already have there with the `external_single_structure(...)`.

Reply:
found:
 public static function execute_returns() {     
        return new external_multiple_structure(
            new external_single_structure( [

Reply:
now the web service is working !

Reply:
congrats!